central/complianceoperator/v2/report/manager/manager_impl.go

```diff
@@ -264,8 +324,16 @@ func (m *managerImpl) handleReadyScan() {
 			return
 		default:
 			if scanResult := m.readyQueue.PullBlocking(m.stopper.LowLevel().GetStopRequestSignal()); scanResult != nil {
-				log.Infof("Scan %s done with %d checks", scanResult.Scan.GetScanName(), len(scanResult.CheckResults))
-				w, err := m.getScanConfigWatcher(scanResult.Ctx, scanResult.Scan, m.scanConfigDataStore, m.scanConfigReadyQueue)
+				// At the moment we simply do not start the ScanConfigWatcher if there are errors in the ScanWatchers
+				// There are many reason why a ScanWatcher might fail like for example the Scan was deleted mid-execution
+				// If this happens we will generate many ReportSnapshots with timeouts. Until we implement a way to
+				// distinguish legitimate failures (CO not reporting back) we log the error and avoid creating the Report.
```

Comment:
Right now if we don't received all the check results from a scan we timeout, log the error, and forget about it. In fact, if any cluster fails to report a scan, we log and forget about it.

I was planning on doing more granular here and do partial reporting (if one scan or cluster fails, we still report in the rest of them) but that introduces a lot of complexity and some cases that I'm not sure how to handle. For example, let's say a scan gets deleted mid-execution. We need to make sure the scan watcher stops, otherwise it will timeout and we will have an error in a report, or maybe we want to have this error (?). Since there are a lot of unknowns on how to handle some of this situations, in this PR I decided to go with the approach of logging and not reporting.

Reply:
Created [ROX-26739](https://issues.redhat.com/browse/ROX-26739) to follow-up on this.